crates/inverted/src/lib.rs

```diff
@@ -0,0 +1,191 @@
+#![allow(clippy::len_without_is_empty)]
+
+use base::distance::DistanceKind;
+use base::index::{IndexOptions, SearchOptions};
+use base::operator::Borrowed;
+use base::operator::Operator;
+use base::scalar::{ScalarLike, F32};
+use base::search::{Collection, Element, Payload, Source, Vectors};
+use base::vector::{VectorBorrowed, VectorKind};
+use common::dir_ops::sync_dir;
+use common::json::Json;
+use common::mmap_array::MmapArray;
+use common::remap::RemappedCollection;
+use quantization::{operator::OperatorQuantization, Quantization};
+use storage::{OperatorStorage, Storage};
+
+use std::collections::BTreeMap;
+use std::fs::create_dir;
+use std::path::Path;
+
+pub trait OperatorInverted: Operator + OperatorQuantization + OperatorStorage {}
+
+impl<T: Operator + OperatorQuantization + OperatorStorage> OperatorInverted for T {}
+
+#[allow(dead_code)]
+pub struct Inverted<O: OperatorInverted> {
+    storage: O::Storage,
+    quantization: Quantization<O>,
```

Comment:
```suggestion
```

Reply:
```rust
let mut documents = (0..self.storage.len()).map(|i| Element { distance: F32::zero(), payload: self.payload[i as usize] }).collect::<Vec<_>>();
// ...
documents.sort();
(Vec::new(), Box::new(documents.into_iter())
```

Reply:
I'm ok with that if `reranker` is not compulsory in the implementation (which makes it different from others).